src/Domain/Charity.php

```diff
@@ -87,6 +87,31 @@ class Charity extends SalesforceReadProxy
      */
     private bool $tbgApprovedToClaimGiftAid = false;
 
+    public function __construct(
+        string $charityName,
```

Comment:
this seems like a good and standard way to ensure every charity has a name - but I'm looking at the current version of the code that we had before deployment today (commit # 663b30e6eaec28b ) and I'm confused about how that caused a crash - there doesn't seem to be any opportunity for anything to need to read the charity name in the time between when we instantiate the charity and when we set the name to a non-null string:

![image](https://github.com/thebiggive/matchbot/assets/159481/9b52d9c1-fe8d-478e-ae6a-2575ef3249a5)

`setSalesforceId` didn't do any funny business.

Reply:
Maybe it's related to this - Doctrine somehow creates a proxy object from nothing based on the relationship? https://github.com/doctrine/orm/issues/3945

Reply:
and I think https://github.com/thebiggive/matchbot/blob/70e2cfa735f7aaf39e924069e043f666f19bf24f/src/Domain/FundRepository.php#L66 means a Campaign can get persisted with no charity. In the DB campaign.charity_id is nullable, but in our Campaign php class $charity is non-nullable. So it seams the campaign gets persisted with no charity, and then when it gets loaded up Doctrine hallucinates a Charity with no properties.

So maybe the fix for this actually isn't giving Charity a no-nullable name, its making Campaign enforce that it must have a Charity, via the DB column and a Campaign constructor.